the-blue-alliance-ios/Frameworks/TBAKit/Sources/TBAKit.swift

```diff
@@ -115,61 +115,61 @@ open class TBAKit: NSObject {
         return task
     }
     
-    internal func callObject<T: TBAModel>(method: String, completion: @escaping (T?, Error?) -> ()) -> URLSessionDataTask {
+    internal func callObject<T: TBAModel>(method: String, completion: @escaping (Result<T?, Error>) -> ()) -> URLSessionDataTask {
         return callApi(method: method) { (response, json, error) in
-            var model: T?
-            if let json = json as? [String: Any] {
-                model = T(json: json)
+            if let error = error {
+                completion(.failure(error))
+            } else if let statusCode = response?.statusCode, statusCode == 304 {
+                completion(.success(nil))
+            } else if let json = json as? [String: Any] {
+                completion(.success(T(json: json)))
+            } else {
+                completion(.failure(APIError.error("Unexpected response from server.")))
             }
-            completion(model, error)
         }
     }
     
-    internal func callArray<T: TBAModel>(method: String, completion: @escaping ([T]?, Error?) -> ()) -> URLSessionDataTask {
+    internal func callArray<T: TBAModel>(method: String, completion: @escaping (Result<[T], Error>) -> ()) -> URLSessionDataTask {
         return callApi(method: method) { (response, json, error) in
-            var models: [T]?
-            if let json = json as? [[String: Any]] {
-                models = []
-                for result in json {
-                    if let model = T(json: result) {
-                        models!.append(model)
-                    }
-                }
+            if let error = error {
+                completion(.failure(error))
+            } else if let statusCode = response?.statusCode, statusCode == 304 {
+                completion(.success([]))
```

Comment:
This is going to be problematic - an empty (null) response is different then a successful empty response and a cached empty response.

I thought https://github.com/the-blue-alliance/the-blue-alliance-ios/issues/336 got closed out, but I guess not. We should just throw this boolean on the responses. Also - we'll need to send etag information for Cloudflare, but that's already half done in dev.

Anywhere we're calling `setLastModified` directly we probably shouldn't. But it's kinda not worth passing the whole response object back up the chain to have TBA send it back down to TBAKit... will need to think about this one.

Reply:
Also keep the Teams view in mind on this one. We’ll bail early on a refresh if we’re not careful.